node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -65,19 +69,23 @@ import scala.util._
 case class P2PClientActor(
     peer: Peer,
     initPeerMsgHandlerReceiver: PeerMessageReceiver,
-    onReconnect: () => Future[Unit]
+    onReconnect: Peer => Future[Unit],
+    onStop: Peer => Future[Unit],
+    maxReconnectionTries: Int
 )(implicit config: NodeAppConfig)
     extends Actor
     with P2PLogger {
 
   private var currentPeerMsgHandlerRecv = initPeerMsgHandlerReceiver
 
-  private var reconnectHandlerOpt: Option[() => Future[Unit]] = None
-
-  private val maxReconnectionTries = 16
+  private var reconnectHandlerOpt: Option[Peer => Future[Unit]] = None
 
+  //total
   private var reconnectionTry = 0
 
+  //from the last initialization
+  private var curReconnectionTry = 0
```

Comment:
Keeping both the total and the current since last initialization. Whenever we reconnect successfully only the current is set to 0 (and used for exponential delays in connection). While the total is used for comparing with max tries.

This fixes an issue where a peer intializes but then disconnects us again and again. Earlier this would be an infinite loop.